flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/GenericHiveMetastoreCatalog.java

```diff
@@ -95,144 +106,103 @@ public void alterDatabase(String name, CatalogDatabase newDatabase, boolean igno
+				.map(fs -> fs.getName())
+				.collect(Collectors.toList());
 		}
-	}
 
-	@Override
-	public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)
-			throws TableNotExistException, CatalogException {
-		if (!tableExists(tablePath)) {
-			if (!ignoreIfNotExists) {
-				throw new TableNotExistException(catalogName, tablePath);
-			}
+		if (TableType.valueOf(hiveTable.getTableType()) == TableType.VIRTUAL_VIEW) {
+			return new GenericCatalogView(
+				hiveTable.getViewOriginalText(),
+				hiveTable.getViewExpandedText(),
+				tableSchema,
+				properties,
+				comment
+			);
 		} else {
-			// IMetastoreClient.alter_table() requires the table to have a valid location, which it doesn't in this case
-			// Thus we have to translate alterTable() into (dropTable() + createTable())
-			dropTable(tablePath, false);
-			try {
-				createTable(tablePath, newTable, false);
-			} catch (TableAlreadyExistException | DatabaseNotExistException e) {
-				// These exceptions wouldn't be thrown, unless a concurrent operation is triggered in Hive
-				throw new CatalogException(
-					String.format("Failed to alter table %s", tablePath), e);
-			}
+			return new GenericCatalogTable(
+				tableSchema, partitionKeys, properties, comment);
 		}
 	}
 
 	@Override
-	public List<String> listTables(String databaseName) throws DatabaseNotExistException, CatalogException {
-		try {
-			return client.getAllTables(databaseName);
-		} catch (UnknownDBException e) {
-			throw new DatabaseNotExistException(catalogName, databaseName);
-		} catch (TException e) {
-			throw new CatalogException(
-				String.format("Failed to list tables in database %s", databaseName), e);
-		}
-	}
+	public Table createHiveTable(ObjectPath tablePath, CatalogBaseTable table) {
+		Map<String, String> properties = new HashMap<>(table.getProperties());
 
-	@Override
-	public List<String> listViews(String databaseName) throws DatabaseNotExistException, CatalogException {
-		try {
-			return client.getTables(
-				databaseName,
-				null, // table pattern
-				TableType.VIRTUAL_VIEW);
-		} catch (UnknownDBException e) {
-			throw new DatabaseNotExistException(catalogName, databaseName);
-		} catch (TException e) {
-			throw new CatalogException(
-				String.format("Failed to list views in database %s", databaseName), e);
-		}
-	}
+		// Table comment
+		properties.put(HiveTableConfig.TABLE_COMMENT, table.getComment());
 
-	@Override
-	public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistException, CatalogException {
-		Table hiveTable = getHiveTable(tablePath);
+		Table hiveTable = new Table();
+		hiveTable.setDbName(tablePath.getDatabaseName());
+		hiveTable.setTableName(tablePath.getObjectName());
+		hiveTable.setCreateTime((int) (System.currentTimeMillis() / 1000));
 
-		return GenericHiveMetastoreCatalogUtil.createCatalogTable(hiveTable);
-	}
+		// Table properties
+		hiveTable.setParameters(buildFlinkProperties(properties));
```

Comment:
Nit: remove the leading spaces.

Reply:
seems to be a github presentation issue, the code doesn't have spaces